pkg/deploy/lattice/target_group_synthesizer_test.go

```diff
@@ -751,3 +785,137 @@ func Test_SynthesizeTriggeredService(t *testing.T) {
 
 	}
 }
+
+func Test_SynthesizeTriggeredTargetGroupsDeletion_TriggeredByK8sService(t *testing.T) {
```

Comment:
Today, K8S service should `not` trigger target group deletion.

The target group is creation is triggered by HTTPRoute's backendref.  Then, it should be deleted by HTTPRoute deletion or not reference this target group.

Reply:
Let me change the each test case name, I intend to test:
For `name: "k8sService triggered target group deletion, ok case"`
it means `name: "httproute deletion with backendref k8sService triggers target group deletion, ok case"`

Reply:
> deleted by HTTPRoute deletion: 

targetGroupSynthesizer.SynthesizeTriggeredTargetGroupsDeletion(ctx) will handle it

> not reference this target group.

targetGroupSynthesizer.SynthesizeSDKTargetGroups(ctx) will handle it